src/python/gudhi/dtm_rips_complex.py

```diff
@@ -12,40 +12,37 @@
 from gudhi.point_cloud.dtm import DistanceToMeasure
 from scipy.spatial.distance import cdist
 
+
 class DTMRipsComplex(WeightedRipsComplex):
     """
-    Class to generate a DTM Rips complex from a distance matrix or a point set, 
+    Class to generate a DTM Rips complex from a distance matrix or a point set,
     in the way described in :cite:`dtmfiltrations`.
-    Remark that all the filtration values are doubled compared to the definition in the paper 
+    Remark that all the filtration values are doubled compared to the definition in the paper
     for the consistency with RipsComplex.
     :Requires: `SciPy <installation.html#scipy>`_
     """
-    def __init__(self, 
-                 points=None, 
-                 distance_matrix=None, 
-                 k=1, 
-                 q=2,
-                 max_filtration=float('inf')):
+
+    def __init__(self, points=None, distance_matrix=None, k=1, q=2, max_filtration=float("inf")):
         """
         Args:
             points (numpy.ndarray): array of points.
             distance_matrix (numpy.ndarray): full distance matrix.
-            k (int): number of neighbors for the computation of DTM. Defaults to 1, which is equivalent to the usual Rips complex.
+            k (int): number of neighbors for the computation of DTM. Defaults to 1, which is equivalent to the usual
+                Rips complex.
             q (float): order used to compute the distance to measure. Defaults to 2.
-            max_filtration (float): specifies the maximal filtration value to be considered.      
+            max_filtration (float): specifies the maximal filtration value to be considered.
         """
         if distance_matrix is None:
             if points is None:
                 # Empty Rips construction
-                points=[]
-            distance_matrix = cdist(points,points)
+                points = [[]]
```

Comment:
I assume this is to avoid an error with `cdist([],[])`? But with this change, we get a distance matrix `[[0]]` of shape (1,1), which should correspond to the case with 1 point, whereas it would seem better to get an empty matrix (possibly of shape (0,0) if that helps). If points is None, couldn't we skip the call to cdist?

Reply:
Oh yes this is better this way, done on [98e1ce1](https://github.com/GUDHI/gudhi-devel/pull/999/commits/98e1ce14bd2ddb3475d06f88b3c61c3b846afabc)